Mathlib/RingTheory/Flat/Basic.lean

```diff
@@ -393,6 +398,22 @@ theorem iff_rTensor_exact : Flat R M ↔
         Function.Exact f g → Function.Exact (f.rTensor M) (g.rTensor M) :=
   iff_rTensor_exact'.{max u v}
 
+/-- If p and q are submodules of M and N respectively, and M and q are flat,
+then `p ⊗ q → M ⊗ N` is injective. -/
+theorem tensorProduct_mapIncl_injective (p : Submodule R M) (q : Submodule R N)
+    [Flat R M] [Flat R q] : Function.Injective (mapIncl p q) := by
+  rw [mapIncl, ← lTensor_comp_rTensor]
+  exact (lTensor_preserves_injective_linearMap _ q.injective_subtype).comp
+    (rTensor_preserves_injective_linearMap _ p.injective_subtype)
+
+/-- If p and q are submodules of M and N respectively, and N and p are flat,
+then `p ⊗ q → M ⊗ N` is injective. -/
+theorem tensorProduct_mapIncl_injective' (p : Submodule R M) (q : Submodule R N)
+    [Flat R p] [Flat R N] : Function.Injective (mapIncl p q) := by
+  rw [mapIncl, ← rTensor_comp_lTensor]
+  exact (rTensor_preserves_injective_linearMap _ p.injective_subtype).comp
+    (lTensor_preserves_injective_linearMap _ q.injective_subtype)
```

Comment:
@acmepjz maybe you already have these somewhere for linear disjointness?

Reply:
It's very similar to [Submodule.LinearDisjoint.of_le_of_flat_left](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/LinearDisjoint.html#Submodule.LinearDisjoint.of_le_of_flat_left) and [Submodule.LinearDisjoint.of_le_of_flat_right](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/LinearDisjoint.html#Submodule.LinearDisjoint.of_le_of_flat_right) but not completely the same. By the way, in my PR about linear disjoint, the only place mentioning `mapIncl` is [Submodule.mulMap_eq_mul'_comp_mapIncl](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/TensorProduct/Submodule.html#Submodule.mulMap_eq_mul'_comp_mapIncl), I think.

Reply:
Do you think it's a good idea to name them as `of_flat_[left|right]` instead of `'`?

Reply:
Yes! I opt for `of_[left|right]`.